src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -152,13 +152,13 @@ class Request
     protected string $defaultLocale = 'en';
 
     /**
-     * @var array<string, string[]>
+     * @var array<string, string[]>|null
      */
-    protected static array $formats;
+    protected static ?array $formats = null;
 
     protected static ?\Closure $requestFactory = null;
 
-    private string $preferredFormat;
+    private ?string $preferredFormat = null;
```

Comment:
I don't think this one needs to be changed. As it is a private method, we can fully control the lazy initialization and the code handling it was fine (as we use `??=`)

Reply:
we can keep it as is, but then we need to update `getPreferredFormat()` to not assign `null` to this property (changes pushed)

Reply:
ah, I missed that we were sometimes assigning `null`